zipline/extensions.py

```diff
@@ -196,22 +196,22 @@ def load(interface, name):
 
 
 @curry
-def register(interface, name, custom_class):
+def register(interface, name, factory):
     """
-    Registers a class for retrieval by the load method
+    Registers a class factory for retrieval by the load method
```

Comment:
With the changes I'm currently implementing (namely adding the ability for the user to specify `*args` and `**kwargs` when calling the load method, which will return a partial function), this will be a factory for instances or partial instances, meaning that a user can either register a method that fully sets up and returns an instance of the class, or register the class's constructor, which can then take extra arguments.